engine/src/main/java/com/google/android/fhir/sync/FhirDataStore.kt

```diff
@@ -100,6 +102,25 @@ internal class FhirDataStore(context: Context) {
     runBlocking { dataStore.edit { pref -> pref[lastSyncTimestampKey] = datetime.toString() } }
   }
 
+  private val mutex = Mutex()
```

Comment:
This should be a mutex map for each key.  Something like following
```suggestion
  private val mutexMap: Map<String, Mutex>
```
Reason: keys (uniqueWorkName) can be different (example - "onetimesync" and "periodicsync") and we dont want to block either of them.

Reply:
done.